### PR TITLE
Allow user to instrument a library without enabling tracing (using the NoopTracer)

### PR DESCRIPTION
**Pull request by mayurkale22**

### CODE REVIEW

**Comment by mayurkale22** [comment]
#277

File: opencensus/trace/noopspan.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime
+from opencensus.trace.span_context import generate_span_id
+from opencensus.trace.tracers import base
+
+
+class NoopSpan(object):
```
Comment (bogdandrutu) [inline opencensus/trace/noopspan.py]: In java we call this BlankSpan, because people may use this even for a request that they don't want to trace.

Comment (mayurkale22) [reply opencensus/trace/noopspan.py]: Changed NoopSpan to BlankSpan...

**Comment by liyanhui1228** [comment]
Is the `NoopSpan` is an improvement of the `NullContextManager`? If so, we could remove the `NullContextManager` class as it won't be used anywhere else.

**Comment by mayurkale22** [comment]
@liyanhui1228 Thanks for the review

**Comment by odeke-em** [comment]
Thanks for the fix @mayurkale22 and @bogdandrutu  @liyanhui1228 for the review! I just have one minor nit: in the future please refer to issues in the commit message so that Github can auto-close them on merge to the main branch e.g.
```shell
trace: make a NoopTracer that drops spans instead of crashing

....[Rest of message]....

Fixes #277
``` 

that way I'd have gotten the notification 11 days ago on it being closed with a PR instead of almost 2 weeks later :)

**Pull request merged**